docs/adr/0006-server-arch.md

```diff
@@ -0,0 +1,185 @@
+- The resolvers should contain no logic themselves. Rather this is delegated
+  to a `use-case` making the resolvers a thin wrapper around the controller logic.
+- Rationale is that it should be easy to implement another interface on the same
+  logic.
+
+#### Example
+
+```
+const { getSubmissionUseCase, } = require('use-cases')
+
+const resolvers = {
+  Query: {
+    async manuscript(_, { id }, { user }) {
+      return getSubmissionUseCase(id, user)
+    },
+    ...
+  }
+  ...
+```
+
+### Use-Cases
+
+- In general this layer will have an interface that maps one-to-one with the
+  graphQL layer.
+- For a given use-case it will require certain services and config - these
+  dependencies will be created by this layer and injected (when needed) into
+  the aggregates / models.
+
+#### Example
+
+```js
+const config = require('config')
+const Storage = require('MyStorageService')
+const Submission = require('aggregates/Submission')
+
+const User = require('@elifesciences/component-model').UserModel
+const File = require('@elifesciences/component-model').FileModel
+const Team = require('@elifesciences/component-model').TeamModel
+const Manuscript = require('@elifesciences/component-model').ManuscriptModel
+
+async getSubmissionUseCase(manuscriptId, user) => {
+  checkPermissions(id, user)
+  const models = { ... }
+  const services = { Storage, ... }
+  const submission = new Submision(config, models, services)
+  return submission.get(manuscriptId, user).toJSON()
+}
+
+module.exports = {
+  getSubmissionUseCase,
+}
+```
+
+### The Aggregate
+
+- [See DDD_Aggregates](https://www.martinfowler.com/bliki/DDD_Aggregate.html)
+- An aggregate
+
+```js
+// Note all dependencies injected.
+
+class Submission {
+  constructor(config, models, services) {
+    this.config = config
+    this.models = models
+    this.storage = services.Storage
+  }
+
+  async get(id, user) {
+      /*
+      * Here we could get the files' contents, the teams and the manuscript
+      * then stitch them together to send back
+      */
+      return {
+        manuscript: await this.models.Manuscript.find(id),
+        teams: await this.models.Team.findByManuscriptId(id),
+        files: await this.models.File.get(config, fileId, this.services.Storage)
+      }
+    }
+  }
+}
+```
+
+### The Models
+
+- The models are for encapsulating a single are of responsiblity in code for a
+  particlar entity within the system. Examples are, Files, Teams, Users, etc.
+- They delegate accessing the data to/from the database to the underlying
+  data access layer.
+- Use Sequelize?
+
+#### Example
+
+```js
+// Note all dependencies injected.
+
+class File {
+  static async get(config, id, storage) {
+    const datamodel = config.datamodel.File // a bit of dependency inversion?
+    return {
+      FileMeta: await datamodel.get(id)
+      FileContent: await storage.getFile(id)
+    }
+  }
+}
+```
+
+### The Data Access Layer
+
+- This layer exists to abstract the code from the data store.
```

Comment:
Usually the [Repository pattern](https://www.martinfowler.com/eaaCatalog/repository.html) is what people go for to keep the models independent of data access, and it is independently included in the list of DDD tactical patterns. The ORM or other technological constraints may impose Active Records, in that case there's not much use for a Repository,

Reply:
Looks good - let me change to reflect this - thanks :)